### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by htv04**
* Switch out MoonJIT with latest LuaJIT v2.1 commit
* Disable LUA52COMPAT since OG LÖVE doesn't use it (and it causes incompatibilities)
* Use OpenAL from portlibs
* Fix love.joystickadded
* Error screen can now be exited by pressing any button
* No Game can now be exited by pressing any button
* Improve app logo quality
* Add file association config for [nx-hbmenu](https://github.com/switchbrew/nx-hbmenu)
* Update README

### CODE REVIEW

**Comment by p-sam** [comment]
Thanks for the PR, you probably should've reach out to me before firing, since frome a quick glance, most of it seems to either needs changes or to be removed. I'd suggest splitting up in multiple small PRs, and as for the contents I can already tell you a few things:

* You don't need to/should not change the code style, since some lines were written with the idea of being able to pull from upstream with too much conflict
* Restrict the changes to fixes, and the lib maintenance for multiple reasons:
	- This is not really meant to be a LÖVE player, and games also usually need some special care any way and as such the file association, while nice seems out of scope to me
	- I'd still like the error or no game screen to be exited from the standard "confirmation" buttons
	- Because of the previous remarks, there's no meaningful changes in the README left, so you can leave them out too
* You did not use gitrepo to pull LuaJIT, it seems drag n drop'd in

I'll be closing this PR for now, feel free to reach out to me here, but also in DMs, or to make some requests via Issues too, because some parts like lib upgrades seems to be the task of the maintainer, so we can split up the work, without kicking your contributions out

To sum it up, we can probably split it in around 3/4 PRs or separate issues you want resolved:
	- LUAJIT pull + LUA52COMPAT
	- Remove openAL-soft and use the provided portlibs one
	- love.joystickadded compat shim (not really a fix, but I get it that it fixes the behavior ultimately)

I probably can do the libs part so you have less clicking to do
